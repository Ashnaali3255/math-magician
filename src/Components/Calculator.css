button {
  border: none;
  cursor: pointer;
}

.container {
  margin-top: 4rem;
  margin-left: 54%;
  display: flex;
  flex-direction: column;
  max-width: 33.33%;
}

.calscreen {
  background-color: #858693;
  display: flex;
  flex-direction: row-reverse;
  padding-right: 2%;
  height: 4rem;
  align-items: center;
  font-size: 1.5rem;
  font-weight: 500;
  color: #fff;
}

.rows {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
}

.calcbtns {
  display: flex;
  justify-content: space-evenly;
  flex-direction: column;
}

.calc-btn {
  display: flex;
  min-width: calc(25% - 1px);
  height: 4rem;
  background-color: #e0e0e0;
  align-items: center;
  justify-content: center;
  border-right: 1px solid #d1d1d1;
  border-bottom: 1px solid #d1d1d1;
}

.orangebtn {
  background-color: #f5913e;
  border-bottom: 1px solid #d1d1d1;
}

.symbols {
  padding-top: 1px;
  border-top: none;
}

.grey {
  border-bottom: 1px solid #d1d1d1;
}

.zero {
  min-width: calc(50% - 1px);
  text-align: center;
  border-bottom: none;
}

.dot {
  border-bottom: none;
}
