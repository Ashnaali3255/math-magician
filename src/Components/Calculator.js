import './Calculator.css';

function Calculator() {
  return (
    <div className="container">
      <div className="calscreen">0</div>
      <div className="calcbtns">
        <div className="rows">
          <div className="calc-btn grey">AC</div>
          <div className="calc-btn grey">+/-</div>
          <div className="calc-btn grey">%</div>
          <div className="calc-btn orangebtn symbols">รท</div>
        </div>
        <div className="rows">
          <div className="calc-btn grey">7</div>
          <div className="calc-btn grey">8</div>
          <div className="calc-btn grey">9</div>
          <div className="calc-btn orangebtn symbols">x</div>
        </div>
        <div className="rows">
          <div className="calc-btn grey">4</div>
          <div className="calc-btn grey">5</div>
          <div className="calc-btn grey">6</div>
          <div className="calc-btn orangebtn symbols">-</div>
        </div>
        <div className="rows">
          <div className="calc-btn grey">1</div>
          <div className="calc-btn grey">2</div>
          <div className="calc-btn grey">3</div>
          <div className="calc-btn orangebtn symbols">+</div>
        </div>
        <div className="rows">
          <div className="calc-btn grey zero">0</div>
          <div className="calc-btn grey dot">.</div>
          <div className="calc-btn orangebtn symbols">=</div>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
