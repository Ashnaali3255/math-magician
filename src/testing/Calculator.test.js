import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../Components/Calculator';
import '@testing-library/jest-dom';

// This test checks if the calculator renders in the DOM
it('check if the Calculator renders properly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Check if buttons are rendered in the DOM', () => {
  render(<Calculator />);
  expect(screen.getByText('AC')).toBeInTheDocument();
  expect(screen.getByText('9')).toBeInTheDocument();
});
