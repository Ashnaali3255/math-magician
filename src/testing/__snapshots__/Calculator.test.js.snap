// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if the Calculator renders properly 1`] = `
<div
  className="container"
>
  <h2
    className="headingcalc"
  >
    Lets do some Math!
  </h2>
  <div
    className="calscreen"
  >
    0
  </div>
  <div
    className="calcbtns"
  >
    <div
      className="rows"
    >
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calc-btn orangebtn symbols"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="rows"
    >
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calc-btn orangebtn symbols"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="rows"
    >
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calc-btn orangebtn symbols"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="rows"
    >
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calc-btn grey"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calc-btn orangebtn symbols"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="rows"
    >
      <button
        className="calc-btn grey zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calc-btn grey dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calc-btn orangebtn symbols"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
